index.js

```diff
@@ -360,14 +362,18 @@ function buildCode (location) {
 
     code += buildValue(propertyLocation, `obj[${JSON.stringify(key)}]`)
 
-    const defaultValue = schema.properties[key].default
+    let defaultValue = schema.properties[key].default
+    if (isRequired) {
+      defaultValue = defaultValue !== undefined ? defaultValue : schema.properties[key].const
```

Comment:
const shouldn't work as a default value. If you define a schema constant property, it must const value, whatever value the user passes.